src/components/templates/VideoListPage.tsx

```diff
@@ -1,70 +1,37 @@
-import React, { useMemo } from 'react'
+import { useMemo } from 'react'
 import { IVideoInfo } from '../../common/interfaces/IVideoInfo'
 import { VideoList } from '../organisms/videoList/VideoList'
 import { PageTitle } from '../atoms/pageTitle/PageTitle'
-import { useSearchParams } from 'react-router-dom'
-import { Role, roleTypeChecker } from '../../common/constants/enums'
-import { videoListPagesInfo } from '../../assets/pages/pagesInfo'
-import videos from '../../assets/videos'
-import { enumMappings } from '../../common/constants/constants'
+import { useLoaderData } from 'react-router-dom'
+import { VideoPageArgs, getVideoInfoForTitle } from '../../common/utils/utils'
+import { useTheme } from '@mui/system'
 
-type VideoListPageEnum = Role | 'documentary'
 /**
  * @returns Page for showing video lists for Roles (Director & Producer, Director of Photography) and Documentary
  */
 export function VideoListPage() {
-  const [queryParams] = useSearchParams()
-  const pageEnum = getPageEnumFromQuery(queryParams)
+  const [videoPageInfo, pageEnum, pageType] = useLoaderData() as VideoPageArgs
+  const theme = useTheme()
 
   const videosForPage = useMemo(
     () =>
-      videos.filter(getFilterForVideos(pageEnum)).filter(video => !!video.url),
-    [pageEnum]
+      videoPageInfo.videoTitles
+        .map(videoTitle => getVideoInfoForTitle(videoTitle))
+        .filter((video): video is NonNullable<IVideoInfo> => !!video),
+    [videoPageInfo.videoTitles]
   )
 
-  if (!pageEnum) {
-    // TODO reroute to home page
-    return null
-  }
+  const titleUnderlineColor = theme.palette[pageEnum].main
 
   return (
     <>
       <PageTitle
-        title={enumMappings[pageEnum].label}
-        subheaderText={videoListPagesInfo[pageEnum].subheaderText}
+        title={videoPageInfo.pageTitle}
+        titleUnderlineColor={titleUnderlineColor}
+        subheaderText={videoPageInfo.subheaderText}
         sx={{ padding: '16px 16px 24px' }}
       />
-      <VideoList videos={videosForPage} pageEnum={pageEnum} />
+      <VideoList videos={videosForPage} pageType={pageType} />
     </>
   )
 }
-
-function getPageEnumFromQuery(
-  query: URLSearchParams
-): VideoListPageEnum | null {
-  const queryValue = query.get('filter')
-
-  if (!queryValue) {
-    // No query found
-    return null
-  }
-
-  const sanitzedValue = queryValue.replaceAll('-', '_')
-
-  if (!roleTypeChecker(sanitzedValue) && sanitzedValue !== 'documentary') {
-    return null
-  }
-
-  return sanitzedValue
-}
```

Comment:
No longer need to calculate from route. Loader function is passing all values in